evm/chains/sharding/mainchain_handler/vmc_handler.py

```diff
@@ -0,0 +1,240 @@
+import logging
+
+from eth_utils import (
+    to_checksum_address,
+    to_dict,
+)
+
+from evm.utils.hexadecimal import (
+    decode_hex,
+)
+
+from evm.chains.sharding.mainchain_handler.config import (
+    DEPOSIT_SIZE,
+    GASPRICE,
+    TX_GAS,
+)
+
+from evm.chains.sharding.mainchain_handler.vmc_utils import (
+    create_vmc_tx,
+    get_contract_address_from_contract_tx,
+    get_vmc_json,
+)
+
+
+class VMCHandler:
```

Comment:
I'd like to suggest a different approach now that I'm starting to see how this is coming together.

Instead of writing a class that delegates to a web3 contract instance, I think it might be better to simply subclass the web3 contract class to add the necessary business logic.

```python
from web3.contract import Contract

class VMC(Contract):
    abi = <the-vmc-abi>
    bytecode = <the-vmc-bytecode>

    def sample(self, shard_id, sender_addr=None, gas=TX_GAS):
        txn_params = self.mk_txn_params(sender=sender, gas=gas)
        address_as_hex = self.call(tx_params).sample(shard_id)
        return decode_hex(address_in_hex)
```

I think this results in a cleaner and easier to read API, removing the initialization code.  Thoughts?

Reply:
Cool! I think it is better. Most of the new logics dealing with mainchain will be implemented in MainchainHandler. I think inheritance should be enough for `VMC`. I'm modifying this part. Thank you.